File: spec/java/src/io/kaitai/struct/spec/TestPositionToEnd.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.PositionToEnd;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestPositionToEnd extends CommonSpec {
    @Test
    public void testPositionToEnd() throws Exception {
        PositionToEnd r = PositionToEnd.fromFile(SRC_DIR + "position_to_end.bin");

        assertEquals(r.index().foo(), 0x42);
        assertEquals(r.index().bar(), 0x1234);
    }
}

File: spec/javascript/test_position_to_end.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('PositionToEnd', 'src/position_to_end.bin', function(r) {
    assert.equal(r.index.foo, 0x42);
    assert.equal(r.index.bar, 0x1234);
});
